Source/FieldSolver/FiniteDifferenceSolver/HybridPICModel/HybridPICModel.cpp

```diff
@@ -534,3 +534,156 @@ void HybridPICModel::FillElectronPressureMF (
         });
     }
 }
+
+void HybridPICModel::BfieldEvolveRK (
+        amrex::Vector<std::array< std::unique_ptr<amrex::MultiFab>, 3>>& Bfield,
+        amrex::Vector<std::array< std::unique_ptr<amrex::MultiFab>, 3>>& Efield,
+        amrex::Vector<std::array< std::unique_ptr<amrex::MultiFab>, 3>> const& Jfield,
+        amrex::Vector<std::unique_ptr<amrex::MultiFab>> const& rhofield,
+        amrex::Vector<std::array< std::unique_ptr<amrex::MultiFab>, 3>> const& edge_lengths,
+        amrex::Real const dt, DtType dt_type,
+        IntVect ng, std::optional<bool> nodal_sync )
+{
+    auto& warpx = WarpX::GetInstance();
+    for (int lev = 0; lev <= warpx.finestLevel(); ++lev)
+    {
+        BfieldEvolveRK(
+            Bfield, Efield, Jfield, rhofield, edge_lengths, dt, lev, dt_type,
+            ng, nodal_sync
+        );
+    }
+}
+
+void HybridPICModel::BfieldEvolveRK (
+        amrex::Vector<std::array< std::unique_ptr<amrex::MultiFab>, 3>>& Bfield,
+        amrex::Vector<std::array< std::unique_ptr<amrex::MultiFab>, 3>>& Efield,
+        amrex::Vector<std::array< std::unique_ptr<amrex::MultiFab>, 3>> const& Jfield,
+        amrex::Vector<std::unique_ptr<amrex::MultiFab>> const& rhofield,
+        amrex::Vector<std::array< std::unique_ptr<amrex::MultiFab>, 3>> const& edge_lengths,
+        amrex::Real const dt, int lev, DtType dt_type,
+        IntVect ng, std::optional<bool> nodal_sync )
+{
+    // Make copies of the B-field multifabs at t = n and create multifabs for
+    // each direction to store the Runge-Kutta intermediate terms. Each
+    // multifab has 2 components for the different terms that need to be stored.
+    std::array< MultiFab, 3 > B_old;
+    std::array< MultiFab, 3 > K;
+    for (int ii = 0; ii < 3; ii++)
+    {
+        B_old[ii] = MultiFab(
+            Bfield[lev][ii]->boxArray(), Bfield[lev][ii]->DistributionMap(), 1,
+            Bfield[lev][ii]->nGrowVect()
+        );
+        MultiFab::Copy(B_old[ii], *Bfield[lev][ii], 0, 0, 1, ng);
+
+        K[ii] = MultiFab(
```

Comment:
If I am not mistaken, then you can reduce the memory footprint of the currently allocated MultiFabs if you destruct the prior stored on before constructing the next one... cc @WeiqunZhang

Reply:
`K[ii]` is empty before the assignment. In some other situations, yes.

Reply:
@ax3l Just to make sure I understand, you mean if `K[ii]` already held a MultiFab it would be better to destruct it before overwriting with a new MultiFab?